src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -17,28 +24,172 @@ class GaussianIVPFilter(ODESolver):
 
     Parameters
     ----------
-    gaussfilt : gaussianfilter.GaussianFilter
-        e.g. the return value of ivp_to_ukf(), ivp_to_ekf1().
-
-    Notes
-    -----
-    - gaussfilt.dynamicmodel contains the prior,
-    - gaussfilt.measurementmodel contains the information about the ODE right hand side function,
-    - gaussfilt.initialdistribution contains the information about the initial values.
+    ivp
+        Initial value problem to be solved.
+    prior
+        Prior distribution.
+    measurement_model
+        ODE measurement model.
+    with_smoothing
+        To smooth after the solve or not to smooth after the solve.
+    init_implementation
+        Initialization algorithm. Either via Scipy (``initialize_odefilter_with_rk``) or via Taylor-mode AD (``initialize_odefilter_with_taylormode``).
+        For more convenient construction, consider :func:`GaussianIVPFilter.from_rk_init`.
     """
 
-    def __init__(self, ivp, gaussfilt, with_smoothing):
-        if not isinstance(gaussfilt.dynamics_model, pnfs.statespace.Integrator):
+    def __init__(
+        self,
+        ivp: IVP,
+        prior: pnfs.statespace.Integrator,
+        measurement_model: pnfs.statespace.DiscreteGaussian,
+        with_smoothing: bool,
+        init_implementation: typing.Callable[
+            [typing.Callable, np.ndarray, float, pnfs.statespace.Integrator], Normal
+        ],
+    ):
+
+        initrv = Normal(
+            np.zeros(prior.dimension),
+            np.zeros((prior.dimension, prior.dimension)),
```

Comment:
Shouldn't this be a standard normal? I'm even surprised that this worked^^

Reply:
It's not really used, but merely a placeholder so that a Kalman object can be initialised. How should I solve this? Put a comment here? Change zeros to NaNs and comment this here?

Reply:
Ok, wondering about this, commenting just as a reminder but I'll come by to discuss